torch/_inductor/codegen/cpp_bmm_template.py

```diff
@@ -83,6 +79,12 @@ def __init__(
         should_block_weights: bool = False,
         name="bmm",
     ):
+        """
+        In order to simplify the implementation and increase code reuse, the BMM template implements
+        two versions of the GEMM kernel: a single-threaded version and a multi-threaded version.
+        GEMM kernels are called in a loop over the batch dimension, with single-threaded GEMM calls
+        for all but the last (B % num_threads), which are handled by the multi-threaded GEMM kernel.
```

Comment:
Please also add note on how to add an extra sizevar arg to the base template etc.